src/main/java/io/cdap/plugin/successfactors/source/input/SuccessFactorsInputFormat.java

```diff
@@ -0,0 +1,105 @@
+import io.cdap.plugin.successfactors.source.transform.SuccessFactorsRecordReader;
+import io.cdap.plugin.successfactors.source.transport.SuccessFactorsTransporter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.olingo.odata2.api.edm.Edm;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This {@code SuccessFactorsInputFormat} defines the InputFormat implementation for SuccessFactors.
+ */
+public class SuccessFactorsInputFormat extends InputFormat<LongWritable, StructuredRecord> {
+
+  public static final String SUCCESSFACTORS_PLUGIN_PROPERTIES = "successFactorsPluginProperties";
+  public static final String PARTITIONS_PROPERTY = "partitionProperty";
+  public static final String ENCODED_ENTITY_METADATA_STRING = "encodedMetadataString";
+  public static final String SERVER_SIDE = "serverSide";
+  private static final Gson GSON = new Gson();
+  private static final Type INPUT_SPLIT_TYPE = new TypeToken<List<SuccessFactorsInputSplit>>() {
+
+  }.getType();
+
+  @Override
+  public List<InputSplit> getSplits(JobContext jContext) throws IOException, InterruptedException {
+
+    Configuration configuration = jContext.getConfiguration();
+    List<InputSplit> splits = new ArrayList<>();
+
+    // Deserialize partitions from Hadoop Configuration
+    List<SuccessFactorsInputSplit> partitions = GSON.fromJson(configuration.get(PARTITIONS_PROPERTY),
+                                                              INPUT_SPLIT_TYPE);
+
+    splits.addAll(partitions);
+
+    return splits;
+  }
+
+
+  @Override
+  public RecordReader<LongWritable, StructuredRecord> createRecordReader(InputSplit split,
+                                                                         TaskAttemptContext taContext)
+    throws IOException, InterruptedException {
+
+    SuccessFactorsInputSplit inputSplit = (SuccessFactorsInputSplit) split;
+
+    SuccessFactorsPluginConfig pluginConfig = GSON.fromJson(taContext.getConfiguration()
+      .get(SUCCESSFACTORS_PLUGIN_PROPERTIES), SuccessFactorsPluginConfig.class);
+
+    Schema outputSchema = Schema.parseJson(taContext.getConfiguration().get(SuccessFactorsSource.OUTPUT_SCHEMA));
+
+    String encodedMetadataString = taContext.getConfiguration().get(ENCODED_ENTITY_METADATA_STRING);
+
+    SuccessFactorsTransporter transporter = new SuccessFactorsTransporter(pluginConfig.getUsername(),
+                                                                          pluginConfig.getPassword());
+
+    SuccessFactorsService successFactorsService = new SuccessFactorsService(pluginConfig, transporter);
```

Comment:
Write a separate method to get SuccessFactorsService and reuse it everywhere.

Reply:
Created a separate method : c60d7165923dc55eab2e5a3dcc247cd770075b47